include/dr/mhp/containers/distributed_vector.hpp

```diff
@@ -128,12 +128,14 @@ class distributed_vector {
   }
 
   ~distributed_vector() {
-    fence();
-    active_wins().erase(win_.mpi_win());
-    win_.free();
-    allocator_.deallocate(data_, data_size_);
-    data_ = nullptr;
-    delete halo_;
+    if (!finalized()) {
```

Comment:
A destructor should clean up the resources. For example the unit tests create many distributed_vectors.

Reply:
Nevermind, I got the logic wrong. I see that this change makes it possible to put the distributed_vector in the same scope as the finalize.